File: VDS_Frontend/src/app/usuarios/Models/usuario.dto.ts
export interface Usuario {
    id?: number;
    numIdent: string;
    numColegiado?: string;
    nombre: string;
    apellido1: string;
    apellido2?: string;
    direccion?: string;
    telefono?: string;
    email?: string;
    rol: Rol;
    username: string;
    password: string;
    clinicaId?: string; // Clínica asociada (solo para veterinarios o admin de clínica
    //CAMBIAR A CLINICA
  }
  
  export enum Rol {
    ADMIN = 'ADMIN',
    ADMIN_CLINICA = 'ADMIN_CLINICA',
    VETERINARIO = 'VETERINARIO',
    CLIENTE = 'CLIENTE'
  }
  

File: VDS_Backend/src/main/java/com/uoc/tfm/vds_backend/usuario/dto/UsuarioDTO.java
package com.uoc.tfm.vds_backend.usuario.dto;

import java.util.List;

import lombok.Data;

@Data
public class UsuarioDTO {
    private Long id;
    private String numIdent;
    private String numColegiado;
    private String nombre;
    private String apellido1;
    private String apellido2;
    private String direccion;
    private String telefono;
    private String email;
    private String rol;
    private String username;
    private String password;
    private Long clinicaId;
    private List<Long> mascotaIds;
}
